src/window.rs

```diff
@@ -1,11 +1,12 @@
 use crate::drm::DrmDisplay;
 use drm::control::{framebuffer::Handle as FramebufferHandle, Device as ControlDevice};
-use gbm::{BufferObjectFlags, Format as BufferFormat, Surface};
+use gbm::{BufferObject, BufferObjectFlags, Format as BufferFormat, Surface};
 
 pub struct Window {
     pub(crate) gbm_surface: Surface<FramebufferHandle>,
     pub(crate) drm_display: DrmDisplay,
-    pub(crate) crtc_set: bool,
+    pub(crate) frame_count: usize,
```

Comment:
Do you think we could get by without storing `frame_count`, and instead just check the presence or absence of `previous_bo`?

Reply:
We could but it would be even better to use a more detailed state enum that tracks whether we have already set a mode, scheduled page flips, etc.

Reply:
Agreed, a state enum would be cleanest.